packages/block-editor/src/components/global-styles/dimensions-panel.js

```diff
@@ -0,0 +1,627 @@
+	const resetPaddingValue = () => setPaddingValues( undefined );
+	const onMouseOverPadding = () => onVisualize( 'padding' );
+
+	// Margin
+	const showMarginControl = useHasMargin( settings );
+	const rawMargin = decodeValue( inheritedValue?.spacing?.margin );
+	const marginValues = splitStyleValue( rawMargin );
+	const marginSides = Array.isArray( settings?.spacing?.margin )
+		? settings?.spacing?.margin
+		: settings?.spacing?.margin?.sides;
+	const isAxialMargin =
+		marginSides &&
+		marginSides.some( ( side ) => AXIAL_SIDES.includes( side ) );
+	const setMarginValues = ( newMarginValues ) => {
+		const margin = filterValuesBySides( newMarginValues, marginSides );
+		onChange( {
+			...value,
+			spacing: {
+				...value?.spacing,
+				margin,
+			},
+		} );
+	};
+	const hasMarginValue = () =>
+		!! value?.spacing?.margin &&
+		Object.keys( value?.spacing?.margin ).length;
+	const resetMarginValue = () => setMarginValues( undefined );
+	const onMouseOverMargin = () => onVisualize( 'margin' );
+
+	// Block Gap
+	const showGapControl = useHasGap( settings );
+	const gapValue = decodeValue( inheritedValue?.spacing?.blockGap );
+	const gapValues = splitGapValue( gapValue );
+	const gapSides = Array.isArray( settings?.spacing?.blockGap )
+		? settings?.spacing?.blockGap
+		: settings?.spacing?.blockGap?.sides;
+	const isAxialGap =
+		gapSides && gapSides.some( ( side ) => AXIAL_SIDES.includes( side ) );
+	const setGapValue = ( newGapValue ) => {
+		onChange( {
+			...value,
+			spacing: {
+				...value?.spacing,
+				blockGap: newGapValue,
+			},
+		} );
+	};
+	const setGapValues = ( nextBoxGapValue ) => {
+		if ( ! nextBoxGapValue ) {
+			setGapValue( null );
+		}
+		// If axial gap is not enabled, treat the 'top' value as the shorthand gap value.
+		if ( ! isAxialGap && nextBoxGapValue?.hasOwnProperty( 'top' ) ) {
+			setGapValue( nextBoxGapValue.top );
+		} else {
+			setGapValue( {
+				top: nextBoxGapValue?.top,
+				left: nextBoxGapValue?.left,
+			} );
+		}
+	};
+	const resetGapValue = () => setGapValue( undefined );
+	const hasGapValue = () => !! value?.spacing?.blockGap;
+
+	// Min Height
+	const showMinHeightControl = useHasMinHeight( settings );
+	const minHeightValue = decodeValue( inheritedValue?.dimensions?.minHeight );
+	const setMinHeightValue = ( newValue ) => {
+		onChange( {
+			...value,
+			dimensions: {
+				...value?.dimensions,
+				minHeight: newValue,
+			},
+		} );
+	};
+	const resetMinHeightValue = () => {
+		setMinHeightValue( undefined );
+	};
+	const hasMinHeightValue = () => !! value?.dimensions?.minHeight;
+
+	// Child Layout
+	const showChildLayoutControl = useHasChildLayout( settings );
+	const childLayout = inheritedValue?.layout;
+	const { orientation = 'horizontal' } = settings?.parentLayout ?? {};
+	const childLayoutOrientationLabel =
+		orientation === 'horizontal' ? __( 'Width' ) : __( 'Height' );
+	const setChildLayout = ( newChildLayout ) => {
+		onChange( {
+			...value,
+			layout: {
+				...layout,
```

Comment:
It looks like this one should be `childLayout` as `layout` appears to be an SVG element. Would it be worth importing `layout` as `layoutIcon` so that it's harder to accidentally reuse `layout`?